opensafely/codelists.py

```diff
@@ -63,39 +68,91 @@ def main():
     pass
 
 
+def add(codelist_url, codelists_dir=None):
+    if not codelists_dir:
+        codelists_dir = Path.cwd() / CODELISTS_DIR
+
+    codelists_file = get_codelist_file(codelists_dir)
+
+    if OPENCODELISTS_BASE_URL not in codelist_url:
```

Comment:
Can we fail earlier with this check? I guess we should have a test for it as well?

Reply:
Yes, and yes!